scripts/invariant_checks/checks/missing-tls-block.js

```diff
@@ -0,0 +1,52 @@
+function isMissingTlsBlock(ingress) {
+  return !ingress.spec.hasOwnProperty('tls')
+}
+
+function formatInfo(ingress) {
+  return `${ingress.metadata.namespace} / ${ingress.metadata.name} - Missing TLS Block`
+}
+
+module.exports.check = async function check(client) {
+  let passed = true;
+  let ingresses = await client.apis.extensions.v1beta1.namespaces('').ingresses.get()
+  for (let ingress of ingresses.body.items) {
+
+    if (isMissingTlsBlock(ingress)) {
+      passed = false;
+      console.log(formatInfo(ingress));
+    }
+  }
+  return passed;
+}
+
+module.exports.fix = async function fix(client) {
+  let ingresses = await client.apis.extensions.v1beta1.namespaces('').ingresses.get()
+  let toFix = ingresses.body.items.filter(isMissingTlsBlock)
+  for (let ingress of toFix) {
+    let spec = ingress.spec;
+    let name = ingress.metadata.name;
+    let namespace = ingress.metadata.namespace;
+    let patch = {
+      body: {
+        spec: {
+          tls: [{
+            hosts: [spec.rules[0].host]
+          }]
+        }
+      }
+    }
+
+    try {
+      let updated = await client.apis.extensions.v1beta1.namespace(
```

Comment:
I'm not familiar with `async`/`await` but it seems quite intuitive and I *think* I get the gist of it.

So the result of patch is async (a Promise) and here we wait for each of them. Potentially we could accumulate these in an array and then loop after this loop to wait for them so potentially they would be submitted concurrently...but to be honest doesn't really matter :P and maybe this is easier to follow.

Reply:
Good point. To be fair, probably these patch operations would be relatively quick anyway so maybe difference is not going to be big, at least at this stage.

As said, no need to change as parallelism is more "confusing".